functions/index.js

```diff
@@ -1,17 +1,139 @@
 const { onSchedule } = require("firebase-functions/v2/scheduler");
 const { setGlobalOptions } = require("firebase-functions/v2");
 const { initializeApp } = require("firebase-admin/app");
+initializeApp();
 const { getFirestore } = require("firebase-admin/firestore");
 const logger = require("firebase-functions/logger");
+const { onDocumentUpdated } = require("firebase-functions/v2/firestore");
```

Comment:
This is really good, it allows for a cleaner and slicker code. Easier to modify for future cases.